### DISCUSSION TITLE
How to perform "group by" queries

### DISCUSSION THREAD

**Discussion by matt-oxley** [supabase/supabase#2246] [Questions] [answered]
Hi there,

I have a working SQL query that relies upon a "GROUP BY" clause:

`SELECT DISTINCT "authorId", count("commitId") FROM commits GROUP BY "authorId" ORDER BY count("commitId") desc`

Is it possible to perform this query using the client API? I can see support for filtering etc. but not for grouping.

**Comment by steve-chavez** [comment, marked as answer]
Hey @matt-oxley,

> Is it possible to perform this query using the client API?

Yes, but not directly. You need to add a view or function. Like:

```sql
create view grouped_commits as
SELECT DISTINCT "authorId", count("commitId") FROM commits GROUP BY "authorId" ORDER BY count("commitId") desc
```

Then call it as:

```js
const { data, error } = await supabase
  .from('grouped_commits')
  .select()
```

**Reply by matt-oxley** [reply]
thanks!
